metron-interface/metron-rest/src/main/java/org/apache/metron/rest/service/GrokService.java

```diff
@@ -0,0 +1,164 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.metron.rest.service;
+
+import oi.thekraken.grok.api.Grok;
+import oi.thekraken.grok.api.Match;
+import org.apache.hadoop.fs.Path;
+import org.apache.metron.common.configuration.SensorParserConfig;
+import org.apache.metron.parsers.GrokParser;
+import org.apache.metron.rest.RestException;
+import org.apache.metron.rest.model.GrokValidation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.StringReader;
+import java.util.Map;
+
+@Service
+public class GrokService {
+
+    public static final String GROK_DEFAULT_PATH_SPRING_PROPERTY = "grok.path.default";
+    public static final String GROK_TEMP_PATH_SPRING_PROPERTY = "grok.path.temp";
+    public static final String GROK_CLASS_NAME = GrokParser.class.getName();
+    public static final String GROK_PATH_KEY = "grokPath";
+    public static final String GROK_STATEMENT_KEY = "grokStatement";
+    public static final String GROK_PATTERN_LABEL_KEY = "patternLabel";
+
+    @Autowired
+    private Environment environment;
+
+    @Autowired
+    private Grok commonGrok;
+
+    @Autowired
+    private HdfsService hdfsService;
+
+    public Map<String, String> getCommonGrokPatterns() {
+        return commonGrok.getPatterns();
+    }
+
+    public GrokValidation validateGrokStatement(GrokValidation grokValidation) throws RestException {
+        Map<String, Object> results;
+        try {
+            Grok grok = new Grok();
+            grok.addPatternFromReader(new InputStreamReader(getClass().getResourceAsStream("/patterns/common")));
+            grok.addPatternFromReader(new StringReader(grokValidation.getStatement()));
+            String patternLabel = grokValidation.getStatement().substring(0, grokValidation.getStatement().indexOf(" "));
+            String grokPattern = "%{" + patternLabel + "}";
+            grok.compile(grokPattern);
+            Match gm = grok.match(grokValidation.getSampleData());
+            gm.captures();
+            results = gm.toMap();
+            results.remove(patternLabel);
```

Comment:
I don't think I fully understand this method. Can you explain what is supposed to be in the results part of the `GrokValidation` object. Is it supposed to be the matched results in the sample data? If so I don't think this will ever happen because they are removed at this line. Am I misunderstanding this correctly or am I missing something?